### PR TITLE
feat(resources): support for camunda cluster clients

### PR DESCRIPTION
**Pull request by Sijoma**
Support for cluster clients. 

Console seems to also support renaming the name of the client in-place. However this is not reflected in the external api. 
https://console.cloud.camunda.io/customer-api/openapi/docs/#/

It seems to be a PUT Request with the clientName in the request body. But I guess we should just request that it gets updated in the SwaggerDocs and update the client lib.
https://console.cloud.camunda.io/api/orgs/ed30ece8-f490-4972-a78f-6c2bfa8b8673/clusters/d8d300e8-996f-4278-8b7c-695d1bce6cc3/clients/1353ebe5-35ea-4699-b1e8-9ff4e226e0b1

image

Also changed the conversion of the GenericOpenAPIError  to dereference the pointer. Did something change here? It must've worked before. 
image

### CODE REVIEW

File: internal/provider/camunda_cluster_client_resource.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+				Type: types.StringType,
+			},
+			"name": {
+				MarkdownDescription: "The name of the cluster client",
+				Required:            true,
+				Type:                types.StringType,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.RequiresReplace(),
+				},
+			},
```
[the remark this answers is not in the document]
Comment (Sijoma) [reply internal/provider/camunda_cluster_client_resource.go:51]: Thanks for the link. I played a bit with it and added the length validator (1-50 characters). Also a "no spaces validator". 

I'm not sure if a RegExp validator would be better (or more common). I did not like that route as it reduces readability sometimes and its harder to combine them as you always have to make sure what the regexp really matches. 

I would keep it at that for this PR.

File: internal/provider/camunda_cluster_client_resource.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+			"Unable to create cluster",
+			fmt.Sprintf("Unable to create cluster, got error: %s", err.(*console.GenericOpenAPIError).Body()),
+		)
+		return
+	}
+
+	data.Id = types.String{Value: inline.Uuid}
+	data.ZeebeClientId = types.String{Value: inline.ClientId}
+	data.Secret = types.String{Value: inline.ClientSecret}
+
```
[the remark this answers is not in the document]
Comment (Sijoma) [reply internal/provider/camunda_cluster_client_resource.go:135]: Makes sense. Should I remove the values for now or can we leave that in for the merge and do the required changes later?

[the remark this answers is not in the document]
Comment (multani) [reply internal/provider/camunda_cluster_client_resource.go:140]: Either we don't expose these values at all, or we need to add a read after the creation to fetch the missing data.

Having modification in the plan, just after an initial `terraform apply`, is a bug usually.

File: internal/provider/camunda_cluster_client_resource.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (multani) [reply internal/provider/camunda_cluster_client_resource.go]: We'll probably have to rewrite it to a regex soon anyway, I guess spaces are not the only invalid characters for a client name.

**Comment by Sijoma** [comment]
I've added the second client read operation. It works with an existing cluster. Some request seems to fail when the cluster is not yet created. Maybe we can merge the change you did for the wait until the cluster is created?

**Comment by multani** [comment]
Worked nicely with the changes from master :tada:

**Review by multani — APPROVED** [review]

File: internal/provider/camunda_cluster_client_resource.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+	clientResp, _, err := r.provider.client.ClientsApi.
+		GetClient(ctx, data.ClusterId.Value, inline.ClientId).
+		Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to fetch client details",
+			fmt.Sprintf("Unable to fetch client details, got error got error: %s",
+				err.(*console.GenericOpenAPIError).Body()))
+	}
```
Comment (multani) [inline internal/provider/camunda_cluster_client_resource.go:152]:
```suggestion
	if err != nil {
		resp.Diagnostics.AddError(
			"Unable to fetch client details",
			fmt.Sprintf("Unable to fetch client details, got error got error: %s",
				err.(*console.GenericOpenAPIError).Body()))
		return
	}
```

**Review by multani — CHANGES_REQUESTED** [review]

**Pull request merged**